save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt

```diff
@@ -20,12 +23,40 @@ internal class OrganizationController(private val organizationService: Organizat
      */
     @GetMapping("/get/organization-name")
     fun getOrganizationByName(@RequestParam name: String): Organization =
-            organizationService.findByName(name)
+            organizationService.findByName(name) ?: throw NoSuchElementException("Organization with name [$name] was not found.")
 
     /**
      * @param owner owner name
      * @return a image
      */
     @GetMapping("/avatar")
-    fun avatar(@RequestParam owner: String): ImageInfo? = organizationService.findByName(owner).avatar.let { ImageInfo(it) }
+    fun avatar(@RequestParam owner: String): ImageInfo? = organizationService.findByName(owner)?.avatar.let { ImageInfo(it) }
+
+    /**
+     * @param newOrganization newOrganization
+     * @param authentication an [Authentication] representing an authenticated request
+     * @return response
+     */
+    @PostMapping("/save")
+    fun saveOrganization(@RequestBody newOrganization: Organization, authentication: Authentication): ResponseEntity<String> {
+        val ownerId = (authentication.details as AuthenticationDetails).id
+        val (organizationId, organizationStatus) = organizationService.getOrSaveOrganization(
+            newOrganization.apply {
+                this.ownerId = ownerId
+                this.dateCreated = LocalDateTime.now()
+            }
+        )
+
+        if (organizationStatus == OrganizationSaveStatus.EXIST) {
+            logger.warn("Organization with id = $organizationId already exists")
```

Comment:
```suggestion
            logger.info("Attempt to save an organization with id = $organizationI, but it already exists")
```
I think it shouldn't be a warning, because it is a valid and properly handled case. Otherwise it'll just pollute logs.

Reply:
Ok, agree